x/evm/txs/base/base.go

```diff
@@ -133,7 +133,9 @@ func (tx *Tx) AnalyzeStop(tag string) {
 }
 
 // SaveTx check Tx do not transition state db
-func (tx *Tx) SaveTx(msg *types.MsgEthereumTx) {}
+func (tx *Tx) SaveTx(msg *types.MsgEthereumTx) {
+	tx.StateTransition.Csdb.SetTransactionHash(*tx.StateTransition.TxHash)
```

Comment:
why save checkTx in checkTx? add annotations to explain it and the old annotation should be deleted

Reply:
because csdb.addLog and csdb.getLogs used csdb.thash, when simulate tx we must use log to deal with some logic like vmbridge and ibc erc20. so we must be set transaction hash